src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs

```diff
@@ -1356,158 +1308,19 @@ public static List<(char First, char Last)>? ComputeRanges(ReadOnlySpan<char> se
-                    case '\u0020': return "\0\u0002\0\u0020\u0021";
-                    case '\u0021': return "\0\u0002\0\u0021\u0022";
-                    case '\u0022': return "\0\u0002\0\u0022\u0023";
-                    case '\u0023': return "\0\u0002\0\u0023\u0024";
-                    case '\u0025': return "\0\u0002\0\u0025\u0026";
-                    case '\u0026': return "\0\u0002\0\u0026\u0027";
-                    case '\u0027': return "\0\u0002\0\u0027\u0028";
-                    case '\u0028': return "\0\u0002\0\u0028\u0029";
-                    case '\u0029': return "\0\u0002\0\u0029\u002A";
-                    case '\u002A': return "\0\u0002\0\u002A\u002B";
-                    case '\u002C': return "\0\u0002\0\u002C\u002D";
-                    case '\u002D': return "\0\u0002\0\u002D\u002E";
-                    case '\u002E': return "\0\u0002\0\u002E\u002F";
-                    case '\u002F': return "\0\u0002\0\u002F\u0030";
-                    case '\u0030': return "\0\u0002\0\u0030\u0031";
-                    case '\u0031': return "\0\u0002\0\u0031\u0032";
-                    case '\u0032': return "\0\u0002\0\u0032\u0033";
-                    case '\u0033': return "\0\u0002\0\u0033\u0034";
-                    case '\u0034': return "\0\u0002\0\u0034\u0035";
-                    case '\u0035': return "\0\u0002\0\u0035\u0036";
-                    case '\u0036': return "\0\u0002\0\u0036\u0037";
-                    case '\u0037': return "\0\u0002\0\u0037\u0038";
-                    case '\u0038': return "\0\u0002\0\u0038\u0039";
-                    case '\u0039': return "\0\u0002\0\u0039\u003A";
-                    case '\u003A': return "\0\u0002\0\u003A\u003B";
-                    case '\u003B': return "\0\u0002\0\u003B\u003C";
-                    case '\u003F': return "\0\u0002\0\u003F\u0040";
-                    case '\u0040': return "\0\u0002\0\u0040\u0041";
-                    case '\u005B': return "\0\u0002\0\u005B\u005C";
-                    case '\u005C': return "\0\u0002\0\u005C\u005D";
-                    case '\u005D': return "\0\u0002\0\u005D\u005E";
-                    case '\u005F': return "\0\u0002\0\u005F\u0060";
-                    case '\u007B': return "\0\u0002\0\u007B\u007C";
-                    case '\u007D': return "\0\u0002\0\u007D\u007E";
-                    case '\u007F': return "\0\u0002\0\u007F\u0080";
-                }
-                AddAsciiCharIgnoreCaseEquivalence(c, ref vsb, caseInsensitive);
-            }
-            else if (!ParticipatesInCaseConversion(c))
-            {
-                resultIsCaseInsensitive = false;
-                vsb.Append(c);
-            }
-            else
-            {
-                resultIsCaseInsensitive = true;
-                vsb.Append(char.ToLower(c, caseInsensitive));
-            }
-
+            var vsb = new ValueStringBuilder(stackalloc char[1]);
+            vsb.Append(c);
```

Comment:
We don't need to create a ValueStringBuilder if at this point we're just creating a span that will always have a single character.  It can be:
```C#
return CharsToStringClass(MemoryMarshal.CreateReadOnlySpan(ref c, 1));
```
or
```C#
return CharsToStringClass(stackalloc char[1] { c });
```